### PR TITLE
Add plugin: Scholar

### PR DESCRIPTION
**Pull request by lolipopshock**
# I am submitting a new Community Plugin

## Repo URL

Link to my plugin: https://github.com/lolipopshock/obsidian-scholar

## Release Checklist
- [x] I have tested the plugin on
  - [ ]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [x]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

### CODE REVIEW

**Comment by lolipopshock** [comment]
Fix in https://github.com/lolipopshock/obsidian-scholar/releases/tag/v1.2.1

**Comment by joethei** [comment]
[import * as path from "path";](https://github.com/lolipopshock/obsidian-scholar/blob/80ca9ca68fb8cd25165e9cef575debb2fd585297/src/main.ts#L48), [import { exec } from "child_process";](https://github.com/lolipopshock/obsidian-scholar/blob/80ca9ca68fb8cd25165e9cef575debb2fd585297/src/obsidianScholar.ts#L13)
These imports are only available from a Node.js runtime, which means that this will throw errors on Mobile (that's a problem since you have `isDesktopOnly` marked as `false` in your manifest.json file.

[this.keyListener = async (event: KeyboardEvent) => {](https://github.com/lolipopshock/obsidian-scholar/blob/80ca9ca68fb8cd25165e9cef575debb2fd585297/src/main.ts#L347-L378)
Instead of adding a listener here, I would recommend using scopes.
```ts
this.scope.register(['Shift'], 'Tab', () => {
	// your code here.
})
```

[text: "SemanticScholar Search Results",](https://github.com/lolipopshock/obsidian-scholar/blob/80ca9ca68fb8cd25165e9cef575debb2fd585297/src/main.ts#L405)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

[file = this.app.vault.getAbstractFileByPath(file) as TFile;](https://github.com/lolipopshock/obsidian-scholar/blob/80ca9ca68fb8cd25165e9cef575debb2fd585297/src/obsidianScholar.ts#L126)
Don't do this blindly, always do a `instanceof` check before converting types.

[containerEl.createEl("h2", { text: SETTING_HEADER });](https://github.com/lolipopshock/obsidian-scholar/blob/80ca9ca68fb8cd25165e9cef575debb2fd585297/src/settingsTab.ts#L59)
[Don't include a header with the plugin name in the settings](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.)

**Comment by lolipopshock** [comment]
Thanks, @joethei ! 

https://github.com/lolipopshock/obsidian-scholar/pull/7 fixes the issues you've mentioned above. 

As for this issue: 
> [import * as path from "path";](https://github.com/lolipopshock/obsidian-scholar/blob/80ca9ca68fb8cd25165e9cef575debb2fd585297/src/main.ts#L48), [import { exec } from "child_process";](https://github.com/lolipopshock/obsidian-scholar/blob/80ca9ca68fb8cd25165e9cef575debb2fd585297/src/obsidianScholar.ts#L13)
These imports are only available from a Node.js runtime, which means that this will throw errors on Mobile (that's a problem since you have isDesktopOnly marked as false in your manifest.json file.

I removed `path` since it's not actually used. There's only one desktop specific command that uses `child_process` (open PDF in system app). And I guard that command with a platform checking: https://github.com/lolipopshock/obsidian-scholar/blob/18d46a3e34dc315609b885b3cfb970caf0ccb305/src/obsidianScholar.ts#L268 . Hope that helps!

**Comment by lolipopshock** [comment]
@joethei  in case you missed it

**Comment by liamcain** [comment]
- [modalInstructionsHTML](https://github.com/lolipopshock/obsidian-scholar/blob/82e6424d8380837eb006cd2e33c7d82c3ae03ef1/src/main.ts#L681) no need to generate the HTML yourself, you can just use [setInstructions](https://github.com/obsidianmd/obsidian-api/blob/791214a68d0dc322b88e5abce617bdf603cc2a2d/obsidian.d.ts#L1174)
- [this.keyListener](https://github.com/lolipopshock/obsidian-scholar/blob/82e6424d8380837eb006cd2e33c7d82c3ae03ef1/src/main.ts#L537) you can use Modal.scope instead and just add a scope handler to Tab and Shift Tab

File: community-plugins.json [hunk 1]
```diff
@@ -8952,5 +8952,11 @@
...
     "repo": "solargate/obsidian-inline-encrypter"
-  }
+  },
+  {
+    "id": "scholar",
+    "name": "Scholar",
+    "author": "Shannon Shen",
+    "description": "Manage research library and streamline research workflow.",
+    "repo": "lolipopshock/obsidian-scholar"
 ]
```
Comment (liamcain) [inline community-plugins.json:8962]:
```suggestion
    "repo": "lolipopshock/obsidian-scholar"
}
]
```

**Pull request merged**